src/tools/diagnose/analyze_resource/analyze_resource_aws.js

```diff
@@ -0,0 +1,87 @@
+/* Copyright (C) 2023 NooBaa */
+'use strict';
+
+const AWS = require('aws-sdk');
+const dbg = require('../../../util/debug_module')(__filename);
+dbg.set_process_name('analyze_resource');
+const http_utils = require('../../../util/http_utils');
+const SensitiveString = require('../../../util/sensitive_string');
+const CloudVendor = require('./analyze_resource_cloud_vendor_abstract');
+
+/**
+ * @typedef {{
+*      access_key: SensitiveString | string,
+*      secret_access_key: SensitiveString | string, 
+*      endpoint: string,
+*      signatureVersion: string
+* }} AnalyzeAwsSpec
+*/
+
+class AnalyzeAws extends CloudVendor {
+    constructor(access_key, secret_access_key, endpoint, signatureVersion) {
+        super(); // Constructors for derived classes must contain a 'super' call.
+        const s3_params = {
+            endpoint: endpoint,
+            accessKeyId: access_key instanceof SensitiveString ? access_key.unwrap() : access_key,
+            secretAccessKey: secret_access_key instanceof SensitiveString ? secret_access_key.unwrap() : secret_access_key,
+            s3ForcePathStyle: true,
+            signatureVersion: signatureVersion,
+            httpOptions: {
+                agent: http_utils.get_default_agent(endpoint)
```

Comment:
for non AWS endpoints you need to create `http_utils.get_unsecured_agent`. see block_store_s3.js as an example